src/test/java/com/microsoft/azure/spring/data/cosmosdb/performance/PerformanceCompareIT.java

```diff
@@ -0,0 +1,270 @@
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
+    }
+
+    @Test
+    public void findByFieldTest() {
+        final List<PerfPerson> data = prepareListData(TIMES);
+
+        final String name = data.get(TIMES / 2).getName();
+
+        final long springCost = runFunctionForTimes(TIMES, name, repository::findByName);
+        final long sdkCost = runConsumerForTimes(TIMES, name, sdkService::findByName);
+
+        report.addItem(new PerfItem(OperationType.FIND_BY_FIELD, springCost, sdkCost, TIMES));
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
+    }
+
+    @Test
+    public void countTest() {
+        final long springCost = getSupplier(TIMES, repository::count);
+        final long sdkCost = getSupplier(TIMES, sdkService::count);
+
+        report.addItem(new PerfItem(OperationType.COUNT, springCost, sdkCost, TIMES));
+        assertPerf(springCost, sdkCost, ACCEPTANCE_DIFF);
+    }
+
+
+    /**
+     * Check whether two time cost fall into the acceptable range.
+     * @param timeCostSpring
+     * @param timeCostSdk
+     * @param acceptanceDiff The acceptable diff between two time cost.
+     */
+    private void assertPerf(long timeCostSpring, long timeCostSdk, float acceptanceDiff) {
+        final long diff = timeCostSpring - timeCostSdk;
+        final float actualDiff = (float) diff / timeCostSdk;
+
+        assertThat(actualDiff).isLessThan(acceptanceDiff);
+    }
+
+    private boolean springDeleteAll() {
+        repository.deleteAll();
+        return true;  // To provide return value for Supplier
+    }
+
+    private List<Sort> buildSortList(Sort sort, int times) {
+        final List<Sort> sorts = Lists.newArrayList();
+        for (int i = 0; i < times; i++) {
+            sorts.add(sort);
+        }
+
+        return sorts;
+    }
+
+    private void queryTwoPages(int pageSize) {
+        final Pageable pageable = new DocumentDbPageRequest(0, pageSize, null);
+        final Page<PerfPerson> page = this.repository.findAll(pageable);
+        this.repository.findAll(page.getPageable());
+    }
+
+    private List<PerfPerson> prepareListData(int count) {
+        final List<PerfPerson> personList = PerfDataProvider.getPerfData(count);
+
+        applyInputListFunc(personList, repository::save);
+        applyInputListFunc(personList, sdkService::save);
+
+        return personList;
+    }
+
+
+    private List<Iterable<PerfPerson>> prepareListBatchData(int times, int batchSize) {
+        final List<Iterable<PerfPerson>> personList = PerfDataProvider.getMultiPerfData(times, batchSize);
+
+        applyInputListFunc(personList, repository::saveAll);
+        applyInputListFunc(personList, sdkService::saveAll);
+
+        return personList;
+    }
+
+    private List<Iterable<String>> listBatchIds(int times, int batchSize) {
+        return prepareListBatchData(times, batchSize).stream()
+                .map(iterable -> {
```

Comment:
looks can be done like `.stream().map(Lists::newArrayList).flatmap(List::stream).map(Person::getId()).Collect(Collectors.toList())`